### PR TITLE
Upgrade Karma

### PR DESCRIPTION
**Pull request by Angamanga**
This pull request makes the following changes:
- Updating Karma to latest version
- Updating dependencies to Karma to the latest version for all packages except:
- Karma-webpack, updates to v 3.0.0, the lastest version for our version of Webpack (will be dealt with when upgrading webpack)
- Karma-jasmine/jasmine-core, updates to v2.0.1 and 3.5 + turns off randomly running the tests (new in Jasmine, but our tests are not written to work with that)

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by Angamanga** [comment]
@rowasc This is as far as I have gotten with Karma, the only thing left is the coverage, that do work with NODE_ENV=test, but I am curious if there is a way to set NODE_ENV=test programmatically. It works setting it in the gulp-task, but not sure if that is correct way of doing it?

File: test/karma.conf.js [hunk 1]
```diff
@@ -40,13 +40,11 @@ module.exports = function (config) {
             'test/unit/spec.bundle.js': ['webpack', 'sourcemap'],
             'mocked_backend/**/*.json': ['json_fixtures']
         },
-
         webpack: require('../webpack.test.config'),
-
-        webpackServer: {
-            noInfo: true // prevent console spamming when running in Karma!
+        webpackMiddleware: {
+            // turning off verbose output
```
Comment (rowasc) [inline test/karma.conf.js]: this is a nice change 🎉

File: test/karma.conf.js [hunk 2]
```diff
@@ -59,7 +57,12 @@ module.exports = function (config) {
                 }
             ]
         },
-
+        client: {
+            // Stops Jasmine from randomly executing tests, our tests are not written that way right now
+            jasmine: {
+                random: false
```
Comment (rowasc) [inline test/karma.conf.js]: ah, cool this is what you mentioned a couple weeks ago. Makes sense

**Comment by rowasc** [comment]
@Angamanga there are some conflicts, but I think it's okay apart from that? LMK when you solve them 🎉

**Comment by Angamanga** [comment]
@rowasc Conflicts fixed!

**Pull request merged**